docs/reference/platforms.rst

```diff
@@ -0,0 +1,218 @@
+
+``base``
+~~~~~~~~
+
+.. code-block:: yaml
+
+    base: <base>
+
+``base`` determines the runtime environment for the charm. It's formatted as
+``<distribution>@<series>`` where ``distribution`` is the name of a Linux
+distribution and ``series`` is the release series name.
+
+Supported bases are:
+
+* ``ubuntu@22.04``
+* ``ubuntu@24.04``
+* ``ubuntu@24.10``
+* ``ubuntu@25.04``
+* ``centos@7``
+* ``almalinux@9``
+
+``build-base``
+~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+    build-base: <base>
+
+``build-base`` determines the environment where the charm is built. If
+``build-base`` is not defined, it defaults to the value of ``base``.
+
+Supported build-bases are the same as the supported bases listed above.
+Additionally, ``devel`` can be used to build a charm using the upcoming Ubuntu
+release in development.
+
+Build plans
+-----------
+
+A build plan is a list of what charms Charmcraft will build, the environments
+where the charms will build, and the environments where the charms will run.
+Build plans are determined by the ``platforms``, ``base``, and ``build-base``
+keywords in a ``charmcraft.yaml``. The build plan can be filtered with
+command-line arguments or environment variables.
+
+Consider the following ``charmcraft.yaml`` snippet:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     amd64:
+       build-on: [amd64]
+       build-for: [amd64]
+     amd64-debug:
+       build-on: [amd64]
+       build-for: [amd64]
+     riscv64-cross:
+       build-on: [amd64, riscv64]
+       build-for: [riscv64]
+
+This snippet generates a build plan with 4 items:
+
++--------+---------------+---------------+-----------------+----------------+--------------+
+| number | platform name | build-on arch | build-time base | build-for arch | runtime base |
++========+===============+===============+=================+================+==============+
+| 1      | amd64         | amd64         | Ubuntu 24.04    | amd64          | Ubuntu 24.04 |
++--------+---------------+---------------+-----------------+----------------+--------------+
+| 2      | amd64-debug   | amd64         | Ubuntu 24.04    | amd64          | Ubuntu 24.04 |
++--------+---------------+---------------+-----------------+----------------+--------------+
+| 3      | riscv64-cross | amd64         | Ubuntu 24.04    | riscv64        | Ubuntu 24.04 |
++--------+---------------+---------------+-----------------+----------------+--------------+
+| 4      | riscv64-cross | riscv64       | Ubuntu 24.04    | riscv64        | Ubuntu 24.04 |
++--------+---------------+---------------+-----------------+----------------+--------------+
+
+If Charmcraft executes on an ``riscv64`` system, it filters the build plan to
+only builds with a ``build-on`` of ``riscv64``. This means Charmcraft will only
+build charm #4.
+
+If Charmcraft executes on an ``amd64`` system, it will build charms #1, #2, and
+#3. This can be further filtered with the ``--platform`` argument or the
```

Comment:
```suggestion
If Charmcraft executes on an ``amd64`` system, it will build charms one, two, and
three. This can be further filtered with the ``--platform`` argument or the
```